app/components/code-edit-modal.js

```diff
@@ -1,9 +1,49 @@
 import Component from '@glimmer/component';
-import { restartableTask, timeout } from 'ember-concurrency';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask } from 'ember-concurrency';
 
 export default class CodeEditModal extends Component {
+  @service('form-code-manager') formCodeManager;
+
+  @tracked formCode;
+  @tracked isButtonsDisabled;
+
+  constructor() {
+    super(...arguments);
+    this.formCode = this.formCodeManager.getTtlOfLatestVersion();
+    this.updateButtonDisabledState();
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+
+    this.args.onCodeChange?.(this.formCode);
+  }
+
   handleCodeChange = restartableTask(async (newCode) => {
-    await timeout(500);
-    this.args.onCodeChange?.(newCode);
+    this.formCodeManager.addFormCode(newCode);
+    this.updateButtonDisabledState();
   });
+
+  @action
+  restoreForm() {
+    this.formCode = this.formCodeManager.getTtlOfReferenceVersion();
+    this.formCodeManager.addFormCode(this.formCode);
+    this.updateButtonDisabledState();
+  }
+
+  @action
+  updateForm() {
+    this.formCode = this.formCodeManager.getTtlOfLatestVersion();
+
+    this.args.onCodeChange?.(this.formCode);
+    this.updateButtonDisabledState();
+  }
```

Comment:
Do we want this to only update or actually save the updated code?

Reply:
This should only update the form code and not save it to the database. This is done because when the form code includes an error the original form code is not overwritten. And users can just refresh the page to fix the issue